clients/src/main/resources/common/message/AlterConfigsRequest.json

```diff
@@ -18,7 +18,9 @@
   "type": "request",
   "name": "AlterConfigsRequest",
   // Version 1 is the same as version 0.
-  "validVersions": "0-1",
+  //
+  // Version 2 will always route to the controller for topic resources change.
```

Comment:
Yes, this should be fixed.

Reply:
There's no mention in KIP-590 of treating requests where `shouldValidateOnly is set differently than ones where this is not set.  I don't think we should do this because in the post-KIP-500 world, only the active controller will have the information needed to validate configuration alteration requests.